From 976b592e8dd40a8d865af4647bfc422c32a3361f Mon Sep 17 00:00:00 2001
From: Ram Muthiah <rammuthiah@google.com>
Date: Wed, 22 Jan 2020 23:43:58 -0800
Subject: [PATCH 636/830] ANDROID: cf build: Use merge_configs
X-NVConfidentiality: public

Previously the cuttlefish build was cat'ng the cuttlefish.fragment
with the gki_defconfig. The correct way to merge fragments is to use
the merge_configs script.

Bug: 139431025
Bug: 148247383
Test: Treehugger
Signed-off-by: Ram Muthiah <rammuthiah@google.com>
Change-Id: I0a62e6c5395b6933955894c8d5ad0484f5c7102d
---
 build.config.cuttlefish.aarch64 | 4 ++--
 build.config.cuttlefish.x86_64  | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/build.config.cuttlefish.aarch64 b/build.config.cuttlefish.aarch64
index e170d360c490..9a685095e934 100644
--- a/build.config.cuttlefish.aarch64
+++ b/build.config.cuttlefish.aarch64
@@ -4,8 +4,8 @@
 
 BUILD_INITRAMFS=1
 DEFCONFIG=cf_aarch_64_gki_defconfig
-PRE_DEFCONFIG_CMDS="cat ./common/arch/arm64/configs/gki_defconfig ./common/cuttlefish.fragment > ./common/arch/arm64/configs/${DEFCONFIG};"
-POST_DEFCONFIG_CMDS="rm ./common/arch/arm64/configs/${DEFCONFIG}"
+PRE_DEFCONFIG_CMDS="KCONFIG_CONFIG=${ROOT_DIR}/common/arch/arm64/configs/${DEFCONFIG} ${ROOT_DIR}/common/scripts/kconfig/merge_config.sh -m -r ${ROOT_DIR}/common/arch/arm64/configs/gki_defconfig ${ROOT_DIR}/common/cuttlefish.fragment"
+POST_DEFCONFIG_CMDS="rm ${ROOT_DIR}/common/arch/arm64/configs/${DEFCONFIG}"
 
 # Not saving any kernel images. This build step is meant purely to generate the .kos.
 FILES=""
diff --git a/build.config.cuttlefish.x86_64 b/build.config.cuttlefish.x86_64
index ccad66ebef5a..d6ce5c3a7306 100644
--- a/build.config.cuttlefish.x86_64
+++ b/build.config.cuttlefish.x86_64
@@ -4,8 +4,8 @@
 
 BUILD_INITRAMFS=1
 DEFCONFIG=cf_x86_64_gki_defconfig
-PRE_DEFCONFIG_CMDS="cat ./common/arch/x86/configs/gki_defconfig ./common/cuttlefish.fragment > ./common/arch/x86/configs/${DEFCONFIG};"
-POST_DEFCONFIG_CMDS="rm ./common/arch/x86/configs/${DEFCONFIG}"
+PRE_DEFCONFIG_CMDS="KCONFIG_CONFIG=${ROOT_DIR}/common/arch/x86/configs/${DEFCONFIG} ${ROOT_DIR}/common/scripts/kconfig/merge_config.sh -m -r ${ROOT_DIR}/common/arch/x86/configs/gki_defconfig ${ROOT_DIR}/common/cuttlefish.fragment"
+POST_DEFCONFIG_CMDS="rm ${ROOT_DIR}/common/arch/x86/configs/${DEFCONFIG}"
 
 # Not saving any kernel images. This build step is meant purely to generate the .kos.
 FILES=""
-- 
2.17.1

